webapp/src/app/projects/[projectName]/[languageName]/layout.tsx

```diff
@@ -0,0 +1,27 @@
+import { Box } from '@mui/material';
+
+import Header from '@/components/Header';
+import Main from '@/components/Main';
+import Sidebar from '@/components/Sidebar';
+
+export default function Layout({
+  // children,
+  explorer,
+  messages,
+}: {
+  // children: React.ReactNode;
+  explorer: React.ReactNode;
+  messages: React.ReactNode;
```

Comment:
Ha, it took a long time for me to understand why we can use `React` in tsx without importing it. Finally I found that this is an old undocumented feature of Next:

https://github.com/vercel/next.js/pull/295
https://github.com/vercel/next.js/pull/8350

This is of course not a problem, requires no fix and you can resolve this conversation when you have read it.